bundles/framework/myplaces3/view/PlaceForm.js

```diff
@@ -9,49 +13,82 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplaces3.view.PlaceForm',
-            '  </div>' +
-            '  <div class="field">' +
-            '    <input type="text" data-name="placedesc" placeholder="' + this.loc('placeform.placedesc.placeholder') + '" />' +
-            '  </div>' +
-            '  <div class="field">' +
-            '    <input type="text" data-name="placeAttention" placeholder="' + this.loc('placeform.placeAttention.placeholder') + '"/>' +
-            '  </div>' +
-            '  <div class="field measurementResult"></div>' +
-            '  <div class="field">' +
-            '    <input type="text" data-name="placelink" placeholder="' + this.loc('placeform.placelink.placeholder') + '"/>' +
-            '  </div>' +
-            '  <div class="field">' +
-            '    <input type="text" data-name="imagelink" placeholder="' + this.loc('placeform.imagelink.placeholder') + '"/>' +
-            '  </div>' +
-            '  <div class="field imagePreview">' +
-            '    <label>' + this.loc('placeform.imagelink.previewLabel') + '</label><br clear="all" />' +
-            '    <a class="myplaces_imglink" target="_blank"><img src=""></img></a>' +
-            '  </div>' +
-            '  <div class="field" id="newLayerForm">' +
-            '    <label for="category">' +
-            '      <span>' + this.loc('placeform.category.choose') + '</span>' +
-            '    </label>' +
-            '    <br clear="all" />' +
-            '    <select data-name="category"></select>' +
-            '  </div>' +
-            '</div>'
-        );
         this.templateOption = jQuery('<option></option>');
         this.categoryForm = undefined;
+
+        this.dialog = undefined;
+        this.dialogForm = undefined;
+
+        // Default generic rules
+        this.defaultRules = [
+            {
+                required: false,
+                message: 'Empty field'
+            }
+        ];
+
+        // Rules for description field
+        this.nameRules = [
+            {
+                required: true,
+                message: this.loc('validation.placeName')
+            },
+            () => ({
+                validator: (_, value) => Oskari.util.sanitize(value) === value ? Promise.resolve(value) : Promise.reject(new Error(this.loc('validation.placeNameIllegal')))
+            })
+        ];
+
+        // Rules for description field
+        this.descriptionRules = [
+            {
+                required: false
+            },
+            () => ({
+                validator: (_, value) => {
+                    if (typeof value !== 'undefined'  && value !== '') {
+                        return Oskari.util.sanitize(value) === value ? Promise.resolve(value) : Promise.reject(new Error(this.loc('validation.descIllegal')));
+                    } else {
+                        return Promise.resolve('');
+                    }
+                }
+            })
+        ];
+
+        this.testRules = [
```

Comment:
This is for testing? Consider moving validator rules to different file that can be imported here.

Reply:
Removed for now